src/nvim/ops.c

```diff
@@ -2524,6 +2528,46 @@ static void yank_copy_line(yankreg_T *reg, struct block_def *bd, long y_idx)
   *pnew = NUL;
 }
 
+static void yank_do_autocmd(oparg_T *oap, yankreg_T *reg)
+{
+  char buf[NUMBUFLEN+2];
+
+  if (!has_event(EVENT_TEXTYANKPOST)) {
+    // No autocommand was defined
+    return;
+  }
+
+  // set v:event to a dictionary with information about the yank
+  dict_T *dict = get_vim_var_dict(VV_EVENT);
+
+  // the yanked text
+  list_T *list = list_alloc();
+  for (linenr_T i = 0; i < reg->y_size; i++) {
+    list_append_string(list, reg->y_array[i], -1);
+  }
+  list->lv_lock = VAR_FIXED;
+  dict_add_list(dict, "regcontents", list);
```

Comment:
Note that while list lock will protect list from modifications one can still do `let v:event.regcontents = 0` because it is dictionary that is locked during initialization, dictionary keys created by `dict_add_*` are explicitly created unlocked.